packages/core/test/actions.test.ts

```diff
@@ -847,6 +847,254 @@ describe('entry/exit actions', () => {
 
       expect(actual).toEqual(['child_b', 'b', 'child_a', 'a', 'root']);
     });
+
+    it('an exit action executed when an interpreter gets stopped should receive `xstate.stop` event', () => {
+      let receivedEvent;
+      const machine = createMachine({
+        exit: (_ctx, ev) => {
+          receivedEvent = ev;
+        }
+      });
+
+      const service = interpret(machine).start();
+      service.stop();
+
+      expect(receivedEvent).toEqual({ type: 'xstate.stop' });
+    });
+
+    // https://github.com/statelyai/xstate/issues/2880
+    it('stopping an interpreter that receives events from its children exit handlers should not throw', () => {
+      const child = createMachine({
+        id: 'child',
+        initial: 'idle',
+        states: {
+          idle: {
+            exit: sendParent('EXIT')
+          }
+        }
+      });
+
+      const parent = createMachine({
+        id: 'parent',
+        invoke: child
+      });
+
+      const interpreter = interpret(parent);
+      interpreter.start();
+
+      expect(() => interpreter.stop()).not.toThrow();
+    });
+
+    it('send actions from exit handlers of a stopped child should not be executed', () => {
```

Comment:
This test should still pass, but by virtue of the parent _ignoring_ any events after being stopped, rather than filtering out `action.send` events.

Reply:
I probably agree with this, this would be better. Let's keep the discussion here: https://github.com/statelyai/xstate/pull/3126#discussion_r821626098